contracts/Admin/VBNBAdmin.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@venusprotocol/governance-contracts/contracts/Governance/AccessControlledV8.sol";
+import "./VBNBAdminStorage.sol";
+
+contract VBNBAdmin is ReentrancyGuardUpgradeable, AccessControlledV8, VBNBAdminStorage {
+    using SafeERC20Upgradeable for IWBNB;
+
+    /// @notice address of vBNB
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    VTokenInterface public immutable vBNB;
+
+    /// @notice address of WBNB contract
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IWBNB public immutable WBNB;
+
+    /// @notice Emitted when PSR is updated
+    event ProtocolShareReserveUpdated(
+        IProtocolShareReserve indexed oldProtocolShareReserve,
+        IProtocolShareReserve indexed newProtocolShareReserve
+    );
+
+    /// @notice Emitted reserves are reduced
+    event ReservesReduced(uint256 reduceAmount);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(VTokenInterface _vBNB, IWBNB _WBNB) {
+        require(address(_WBNB) != address(0), "WBNB address invalid");
+        require(address(_vBNB) != address(0), "vBNB address invalid");
+
+        vBNB = _vBNB;
+        WBNB = _WBNB;
+
+        // Note that the contract is upgradeable. Use initialize() or reinitializers
+        // to set the state variables.
+        _disableInitializers();
+    }
+
+    function initialize(
+        IProtocolShareReserve _protocolShareReserve,
+        address accessControlManager
+    ) external initializer {
+        require(address(_protocolShareReserve) != address(0), "PSR address invalid");
+        protocolShareReserve = _protocolShareReserve;
+
+        __ReentrancyGuard_init();
+        __AccessControlled_init(accessControlManager);
+    }
+
+    /**
+     * @dev PSR setter.
+     * @param protocolShareReserve_ Address of the PSR contract
+     */
+    function setProtocolShareReserve(IProtocolShareReserve protocolShareReserve_) external {
+        require(msg.sender == owner(), "only owner can set PSR");
+
+        require(address(protocolShareReserve_) != address(0), "PSR address invalid");
+        emit ProtocolShareReserveUpdated(protocolShareReserve, protocolShareReserve_);
+        protocolShareReserve = protocolShareReserve_;
+    }
+
+    /**
+     * @notice Reduce reserves of vBNB
+     * @param reduceAmount amount of reserves to reduce
+     */
+    function reduceReserves(uint reduceAmount) external nonReentrant {
```

Comment:
Just being curious.

Only authorised users can `VToken::_reduceReserves()`. However, anyone can do `VBNBAdmin::reduceReserves()`. Not sure we have a special consideration?

Reply:
The goal is to make permisionless the "automatic income allocation". `VBNBAdmin::reduceReserves()` is going to transfer the reduced reserves to a pre-fixed address, set by Governance via VIP, so it should be safe and it's not needed to limit it to authorized accounts. VBNB is not upgradable, so the behavior of its `VToken::_reduceReserves()` cannot be changed. We think we are covering the requirements (permissionless) with the proposed solution.